File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        arr.sort()
        v = []

        i = 0
        while i < len(arr):
            cnt = 1

            # Count occurrences of the current element
            while i + 1 < len(arr) and arr[i] == arr[i + 1]:
                cnt += 1
                i += 1

            v.append(cnt)
            i += 1

        v.sort()

        for i in range(1, len(v)):
            if v[i] == v[i - 1]:
                return False

        return True

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.js

var uniqueOccurrences = function(arr) {
    arr.sort((a, b) => a - b);
    let v = [];

    for (let i = 0; i < arr.length; i++) {
        let cnt = 1;

        // Count occurrences of the current element
        while (i + 1 < arr.length && arr[i] === arr[i + 1]) {
            cnt++;
            i++;
        }

        v.push(cnt);
    }

    v.sort((a, b) => a - b);

    for (let i = 1; i < v.length; i++) {
        if (v[i] === v[i - 1]) {
            return false;
        }
    }

    return true;
};

